Sources/Build/misc.swift

```diff
@@ -43,6 +43,42 @@ extension CModule {
     }
 }
 
+extension String {
+    var isCpp: Bool {
+        guard let ext = self.fileExt else {
+            return false
+        }
+        return ext == "cpp"
+    }
+}
+
+extension ClangModule {
+    // Returns language specific arguments for a ClangModule.
+    var languageArgs: [String] {
```

Comment:
This is relatively expensive to crawl over the transitive closure of the dependencies. I am fine accepting this PR to make forward progress, but there should be a FIXME here about ensure we get proper infrastructure to cache this kind of thing.